bootstrap/bootstrap.go

```diff
@@ -85,6 +88,42 @@ func (b *Bootstrap) Run(ctx context.Context) (exitCode int) {
 	defer stopper()
 	defer func() { span.FinishWithError(err) }()
 
+	if experiments.IsEnabled("job-api") {
+		if runtime.GOOS == "windows" {
+			b.shell.Errorf("The job API is not currently supported on Windows. It's on its way though, promise!")
+			return 1
+		}
+
+		jobAPISocketPath := path.Join(os.TempDir(), fmt.Sprintf("buildkite-agent-%d.sock", os.Getpid()))
```

Comment:
I'm a bit worried by `os.TempDir` here: It defaults to `/tmp`, which is usually world-writable, so something else could (usually accidentally) start messing with our socket.

It turns out there's [a standardised place](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s15.html) for Unix sockets: `/run` (or a subdirectory). Using this creates a new problem for us, which is, there might not be an existing subdirectory writable by the agent.